src/Lean/Elab/Tactic/Induction.lean

```diff
@@ -216,7 +216,11 @@ private def checkAltNames (alts : Array Alt) (altsSyntax : Array Syntax) : Tacti
     let altName := getAltName altStx
     if altName != `_ then
       unless alts.any (·.name == altName) do
-        throwErrorAt altStx "invalid alternative name '{altName}'"
+        let validNames := (alts.map (·.name)).toList |>List.take 3
+        let validNamesStr := validNames.foldl (init := "") fun acc n =>
+          if acc == "" then s!"{Name.toString n}" else s!"{acc}, {Name.toString n}"
+        throwErrorAt altStx s!"invalid alternative name '{altName}', valid options include: {validNamesStr}"
```

Comment:
Thanks for looking into this. I suggest to do a bit more while you are at it, and make the `checkAltNames` function check

* That `_` is last, if it exists.
* That no name is duplicate.
* When an name is not found in `alts`, say “invalid alternative name '{altName}', expected one of …, … or, …“, and list all those alternative names that are not explicitly handled already. (If this list is empty, print a different message, e.g. “…, no unhandled alternatives”.

Reply:
>That _ is last, if it exists.

this is done already in the code, right?
```
if getAltName altStx == `_ && i != altsSyntax.size - 1 then
      withRef altStx <| throwError "invalid occurrence of wildcard alternative, it must be the last alternative"
```
happy to tackle the other two suggestions.